### PR TITLE
Add log parsing to system test

### PR DESCRIPTION
**Pull request by raghusarangapani**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:
- Change the log scanning logic from simple if "string" in to take a regex.
- Add log scanning variables to system test.

All existing tests in test_scan_logs.py passed. I also separately verified the scanning logic.

### CODE REVIEW

File: utilities/scan_logs.py [hunk 1]
```diff
@@ -68,7 +69,8 @@ def scan_for_errors(log_file_path, error_strings):
             for word in error_strings:
                 # convert the word to lowercase and the line to all lower case
                 # which handles the case where 'warning' will catch 'WARNING' and 'Warning', etc
-                if word.lower() in line.lower():
+                # if word.lower() in line.lower():
+                if re.search(word.lower(), line.lower()):
```
Comment (sethrosetter) [inline utilities/scan_logs.py]: Can you add some unit tests?

Comment (raghusarangapani) [inline utilities/scan_logs.py]: It is already there, test_scan_logs.py

Comment (raghusarangapani) [inline utilities/scan_logs.py]: This change is to enable scanning for strings with regex, like - "Out of memory: Kill process 1234 (sync_gateway)" as "Out of memory: Kill process \d+ \(sync_gateway\)"

Comment (sethrosetter) [inline utilities/scan_logs.py]: Test for new keywords

File: testsuites/syncgateway/system/conftest.py [hunk 2]
```diff
@@ -180,6 +182,7 @@ def params_from_base_test_setup(request, params_from_base_suite_setup):
 
     # Verify all sync_gateways and sg_accels are reachable
     c = cluster.Cluster(cluster_config)
+    remote_executor = RemoteExecutor(c.sync_gateways[0].ip)
```
Comment (sethrosetter) [inline testsuites/syncgateway/system/conftest.py]: I don't think we can rely on this being a single node. Also. RemoteExecutor will not work with Docker. Ansible will be able to.

File: testsuites/syncgateway/system/conftest.py [hunk 3]
```diff
@@ -183,8 +185,45 @@ def params_from_base_test_setup(request, params_from_base_suite_setup):
...
     assert len(errors) == 0
+
+    # Scan logs
+    # SG logs for panic, data race
+    # System logs for OOM
+    sys_log_file = "/var/log/messages"
+
+    playbook_vars = {
+        "error_string": "Out of memory: Kill process",
+        "secondary_error_string": "sync_gateway",
```
Comment (sethrosetter) [inline testsuites/syncgateway/system/conftest.py]: Will this catch sg_accel OOM?

**Pull request merged**